theano/gpuarray/tests/test_basic_ops.py

```diff
@@ -182,6 +182,19 @@ def test_transfer_cpu_gpu():
     assert numpy.all(fv == av)
 
 
+def test_transfer_gpu_gpu():
+    g = GpuArrayType(dtype='float32', broadcastable=(False, False),
+                     context_name=test_ctx_name)()
+
+    av = numpy.asarray(rng.rand(5, 4), dtype='float32')
+    gv = gpuarray.array(av, context=get_context(test_ctx_name))
+
+    f = theano.function([g], GpuToGpu(test_ctx_name)(g))
```

Comment:
You will need to disable the `local_cut_gpu_transfers` opt because otherwise it only leaves a DeepCopyOp in the graph.  The op itself will work with the same context as source and target.